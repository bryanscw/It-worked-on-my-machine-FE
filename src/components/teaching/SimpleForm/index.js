import React, { Component } from 'react';
import { EMPTY } from '../../../utils/constants';

export class SimpleForm extends Component {
    emptyState = {
        title: '',
        description: ''
    };
    state = this.emptyState;

    componentDidMount() {
        const {
            initialState
        } = this.props;

        if (initialState) {
            if (initialState === EMPTY)
                this.setState(this.emptyState);

            else
                this.setState(initialState);
        }
    }

    componentDidUpdate(prevProps) {
        const {
            initialState
        } = this.props;

        if (initialState && initialState !== prevProps.initialState) {
            if (initialState === EMPTY)
                this.setState(this.emptyState);
                
            else
                this.setState(initialState);
        }
    }

    onSubmit = e => {
        e.preventDefault();

        const {
            title,
            description
        } = this.state;

        this.props.onSubmit({
            title,
            description
        });

        this.setState(this.emptyState);
    }

    onChange = e => {
        this.setState({
            [e.target.name]: e.target.value
        });
    }

    render() {
        const {
            title,
            description
        } = this.state;

        return (
            <form className="form" onSubmit={this.onSubmit}>
                <div className="form-group">
                    <label>Title</label>
                    <input
                        className="form-control"
                        type="text"
                        name="title"
                        onChange={this.onChange}
                        value={title}
                        />
                </div>
                <div className="form-group">
                    <label>Description</label>
                    <textarea
                        className="form-control"
                        name="description"
                        onChange={this.onChange}
                        value={description}
                        />
                </div>
                <div className="form-group mt-4">
                    <button className="btn btn-primary" type="submit">Submit</button>
                </div>
            </form>
        )
    }
}

export default SimpleForm
