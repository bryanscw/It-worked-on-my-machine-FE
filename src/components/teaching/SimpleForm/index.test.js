import React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'
import SimpleForm from './index.js';

afterEach(cleanup);

it('should take a snapshot', () => {
    const { asFragment } = render(<SimpleForm />);
    expect(asFragment()).toMatchSnapshot();
})

it('should return form data', () => {
    const expectedResult = {
        title: 'Title',
        description: 'Description',
    }

    var result;
    const { getByLabelText, getByText } = render(<SimpleForm onSubmit={(formResult) => result = formResult}  />);

    fireEvent.change(getByLabelText(/title/i), {
        target: {value: expectedResult.title},
    })
    fireEvent.change(getByLabelText(/description/i), {
        target: {value: expectedResult.description},
    })
    fireEvent.click(getByText(/submit/i));
    
    expect(result).toEqual(expectedResult);
});