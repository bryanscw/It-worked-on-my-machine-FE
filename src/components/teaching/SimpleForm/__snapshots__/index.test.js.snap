// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <form
    class="form"
  >
    <div
      class="form-group"
    >
      <label
        for="simpleForm-title"
      >
        Title
      </label>
      <input
        class="form-control"
        id="simpleForm-title"
        name="title"
        type="text"
        value=""
      />
    </div>
    <div
      class="form-group"
    >
      <label
        for="simpleForm-description"
      >
        Description
      </label>
      <textarea
        class="form-control"
        id="simpleForm-description"
        name="description"
      />
    </div>
    <div
      class="form-group mt-4"
    >
      <button
        class="btn btn-primary"
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</DocumentFragment>
`;
